packaging/datadog-agent/source/setup_agent.sh

```diff
@@ -389,6 +389,26 @@ tar -xz -C "$DD_HOME/agent" --strip-components 1 -f "$DD_HOME/agent.tar.gz"
 rm -f "$DD_HOME/agent.tar.gz"
 print_done
 
+print_console "* Downloading integrations from GitHub"
+mkdir -p "$DD_HOME/integrations"
+$DOWNLOADER "$DD_HOME/integrations.tar.gz" "https://api.github.com/repos/DataDog/integrations-core/tarball"
+print_done
+
+print_console "* Uncompressing tarball"
+tar -xz -C "$DD_HOME/integrations" --strip-components 1 -f "$DD_HOME/integrations.tar.gz"
+rm -f "$DD_HOME/integrations.tar.gz"
+print_done
+
+print_console "* Trying to install integration requirements"
+for INT_DIR in "$DD_HOME/integrations/*"; do
+  if [ -d $INT_DIR ]; then
+    if [-f "$INT_DIR/requirements.txt" ]; then
+      "$DD_HOME/agent/utils/pip-allow-failures.sh" "$INT_DIR/requirements.txt"
```

Comment:
let's concatenate all the requirement files (agent + integrations) in one constraint file and use that constraint with every pip call so that we're sure we're not installing conflicting dependencies.

We can even use the output of `pip freeze` instead of the agent's `requirements.txt` file to generate that constraint file